### PR TITLE
HTML/CSS Guidelines for planner

### PR DESCRIPTION
**Pull request by SMahil**
***WIP***

 - [ ] What does this PR do? 
This PR will follow the HTML/CSS guidelines given in 
https://github.com/fabric8-ui/fabric8-ui/wiki/CSS-Naming-Conventions

- [ ] What issue/task does this PR references?
https://openshift.io/openshiftio/Openshift_io/plan/detail/103

- [ ] Are the tests Included?

NA
___
### Optional
- [ ] Is the documentation Included?

- [ ] Are the Release Notes included?

- [ ] @mention(s) to expected reviewer(s) included
@vikram-raj

### CODE REVIEW

**Comment by SMahil** [comment]
[test]

**Comment by SMahil** [comment]
[test]

**Comment by SMahil** [comment]
[test]

**Comment by christianvogt** [comment]
The prefix for all planner classes should be qualified with `f8-planner`.

File: src/app/components_ngrx/work-item-event/work-item-event.component.less [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
-    margin-left: 5px;
-    color: @color-pf-black-400;
-    cursor: pointer;
-  }
 }
-.audit-trail__label {
-  display: contents;
+.f8-audit-trail__label {
+  display: inherit;
+  f8-label {
```
Comment (christianvogt) [inline src/app/components_ngrx/work-item-event/work-item-event.component.less]: Missing `.` ? Why do you even have to change the style of a nested component? Seems hackish.

File: src/app/components_ngrx/work-item-event-wrapper/work-item-event-wrapper.component.less [hunk 2]
```diff
@@ -1,10 +1,10 @@
 @import (reference) '../../../assets/stylesheets/_base';
 
 .f8-event {
-  &-content {
+  &__list {
     li {
       text-align: center;
-      &.f8-toggle-caret {
+      &.f8-comment__toggle {
```
Comment (christianvogt) [inline src/app/components_ngrx/work-item-event-wrapper/work-item-event-wrapper.component.less]: This must nesting seems unnecessary when you have a very specific element level class already.

**Comment by Raunak1203** [comment]
[test]

**Comment by SMahil** [comment]
[test]

**Comment by SMahil** [comment]
[test]

File: src/app/components_ngrx/work-item-event-wrapper/work-item-event-wrapper.component.html [hunk 3]
```diff
@@ -1,11 +1,13 @@
-<div class="f8-event" id="wi-event">
-  <ul class="f8-event-content nav nav-tabs nav-tabs-pf" 
+<div class="f8-planner__event-wrapper"
+  id="wi-event">
+  <ul class="f8-planner__event-list nav nav-tabs nav-tabs-pf" 
     id="wi-event-content" 
     *ngIf="true">
     <!-- Refactoring needed to toogle view -->
-    <li class="f8-toggle-caret" 
+    <li class="f8-planner__comment--toggle"
```
Comment (christianvogt) [inline src/app/components_ngrx/work-item-event-wrapper/work-item-event-wrapper.component.html]: Why is this renamed with `comment`, which is a class that exists elsewhere?

File: src/app/components_ngrx/work-item-event/work-item-event.component.less [hunk 4]
```diff
@@ -1,36 +1,35 @@
 @import (reference) '../../../assets/stylesheets/_base';
 
-.f8-event {
-  &__wrap {
+.f8-planner {
+  &__event {
```
Comment (christianvogt) [inline src/app/components_ngrx/work-item-event/work-item-event.component.less]: This should be `f8-planner-event`

File: src/app/components_ngrx/work-item-link/work-item-link.component.less [hunk 5]
```diff
@@ -1,12 +1,11 @@
 @import (reference) '../../../assets/stylesheets/_base';
 
-.f8-link {
-  // &-container { }
-  &__nav-list {
+.f8-planner__link {
```
Comment (christianvogt) [inline src/app/components_ngrx/work-item-link/work-item-link.component.less]: This should be `f8-planner-link`

File: src/app/components_ngrx/work-item-preview-panel/work-item-preview-panel.component.less [hunk 6]
```diff
@@ -1,6 +1,6 @@
 @import (reference) '../../../assets/stylesheets/_base';
 
-.f8-quick-preview {
+.f8-planner__quick-preview {
```
Comment (christianvogt) [inline src/app/components_ngrx/work-item-preview-panel/work-item-preview-panel.component.less]: This should be `f8-planner-quick-preview`

File: src/app/widgets/comment-module/comment.component.less [hunk 7]
```diff
@@ -1,6 +1,6 @@
 @import (reference) '../../../assets/stylesheets/_base';
 
-.f8-comment {
+.f8-planner__comment {
```
Comment (christianvogt) [inline src/app/widgets/comment-module/comment.component.less]: This should be `f8-planner-comment`. Although make sure it doesn't conflict with the other comment class. This is why naming blocks based on the component name is a good practice to follow.

File: src/app/widgets/inlineinput/inlineinput.component.less [hunk 8]
```diff
@@ -1,8 +1,8 @@
 @import (reference) './../../../assets/stylesheets/_base';
 
-.inlineinput-container {
+.f8-inlineinput {
```
Comment (christianvogt) [inline src/app/widgets/inlineinput/inlineinput.component.less]: Missing `planner` domain in selector.

File: src/app/widgets/planner-layout/planner-layout.component.less [hunk 9]
```diff
@@ -1,15 +1,16 @@
...
-  top: 0;
-  bottom: 0;
-  width: 320px;
-  min-height: 100%;
-}
-.content-wrapper {
-  position: relative;
-  min-height: 100%;
-  margin-left: 320px;
+.f8-planner {
```
Comment (christianvogt) [inline src/app/widgets/planner-layout/planner-layout.component.less]: Missing block element.

File: src/app/widgets/inlineinput/inlineinput.component.less (hunk 8 above)
Comment (SMahil) [reply src/app/widgets/inlineinput/inlineinput.component.less]: inline input is from common widgets which can be contributed outside planner. Should we nest these under planner domain?

File: src/app/components_ngrx/planner-board/planner-board.component.less [hunk 10]
```diff
@@ -51,7 +37,13 @@
       }
     }
   }
-}
-.f8-board-column__body {
-  height: 100%;
+  .f8-planner-board {
```
[the remark this answers is not in the document]
Comment (SMahil) [reply src/app/components_ngrx/planner-board/planner-board.component.less]: as f8-planner-container is the outer wrapper

File: src/app/components_ngrx/planner-board/planner-board.component.less [hunk 11]
```diff
@@ -5,23 +5,9 @@
...
-      margin-left: 10px;
-      &:first-of-type {
-        margin-left: 0;
-      }
-    }
-  }
-  .lock-overlay-list {
+
+.f8-planner-container {
+  .f8-planner-overlay {
```
[the remark this answers is not in the document]
Comment (SMahil) [reply src/app/components_ngrx/planner-board/planner-board.component.less]: same as above

**Comment by SMahil** [comment]
[test]

**Comment by vikram-raj** [comment]
screenshot from 2018-09-21 12-42-12

Scrollbar is visible for the column.

**Comment by SMahil** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by joshuawilson** [comment]
@christianvogt do we close this or do you want to update it?

**Comment by christianvogt** [comment]
At this point the PR is too large, out of date, and has outstanding issues to be merged with confidence.

We will have to breakdown the work into smaller, more manageable pieces.

**Comment by joshuawilson** [comment]
based on that recommendation I'm going to close this
